elements/lisk-chain/src/block.ts

```diff
@@ -76,5 +77,26 @@ export class Block {
 		for (const tx of this.payload) {
 			tx.validate();
 		}
+
+		const assets = this.assets.getAllAssets();
+		let last = assets[0];
+		let i = 0;
+		for (const asset of assets) {
+			// Data size of each module should not be greater than max asset data size
+			if (asset.data.byteLength > MAX_ASSET_DATA_SIZE_BYTES) {
```

Comment:
it's good point to  use `byteLength`.  We might need to  check where we use `length` for Buffer and  make it consistent

Reply:
we can create an issue for this or consider this in upcoming PRs